extension/scripts/content.js

```diff
@@ -18,18 +18,21 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
 
 		const observer = new MutationObserver((mutations) => {
 			mutations.forEach((m) => {
-				// console.log('Mutation: ', m.type)
+				// console.log('Mutation className: ', m.addedNodes[0].className)
 				if (
 					m.type === 'childList' &&
 					m.addedNodes.length > 0 &&
+					m.addedNodes[0] &&
 					m.addedNodes[0].className === '' &&
 					m.addedNodes[0].tagName === 'DIV'
 				) {
-					console.log('added:', m.addedNodes)
-					chrome.runtime.sendMessage({
-						content:
-							document.querySelector('div[id="main"]').innerHTML,
-					})
+					setTimeout(() => {
+						chrome.runtime.sendMessage({
+							content:
+								document.querySelector('div[id="main"]')
+									.innerHTML,
+						})
+					}, 240)
```

Comment:
Because the image load depends on whatsapp and also on internet speed, there is not much we can do about it. I tried different solutions to wait out the lazy loading of the final image and this one gave me the best results. Once we notice that something changed on the whatsapp page, we make sure it's in the correct group (if statement) and then wait for the final image to be loaded (timeout).